Kentico/xperience-algolia
# Enable index name customization

### Motivation

Implements #22

### Checklist


### How to test

Manually register Algolia indexes using the new approach described [here](https://github.com/Kentico/xperience-algolia/tree/feat/set-index-names#gear-creating-and-registering-an-algolia-index), e.g.

```cs
protected override void OnPreInit()
{
    base.OnPreInit();

    var environment = ConfigurationManager.AppSettings["Environment"];
    Service.Use<IAlgoliaIndexRegister>(new DefaultAlgoliaIndexRegister()
        .Add<AlgoliaSiteSearchModel>($"{environment}-{AlgoliaSiteSearchModel.IndexName}")
    );
}
```

and ensure that the indexing/search functionality is working as expected. Indexes using the [`RegisterAlgoliaIndex`](https://github.com/Kentico/xperience-algolia/blob/master/src/Attributes/RegisterAlgoliaIndexAttribute.cs) attribute should still work as expected.
